From 4971f443726360216a4ef3ba8baea258a1cd0f3b Mon Sep 17 00:00:00 2001
From: Antonio Alvarez Feijoo <antonio.feijoo@suse.com>
Date: Thu, 27 Jul 2023 11:08:49 +0200
Subject: [PATCH] fix(systemd-journald): add systemd-sysusers dependency

---
 modules.d/01systemd-journald/module-setup.sh | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/modules.d/01systemd-journald/module-setup.sh b/modules.d/01systemd-journald/module-setup.sh
index dd1b3f3bf3..3cf2a1a119 100755
--- a/modules.d/01systemd-journald/module-setup.sh
+++ b/modules.d/01systemd-journald/module-setup.sh
@@ -19,6 +19,8 @@ check() {
 # Module dependency requirements.
 depends() {
 
+    # This module has external dependency on other module(s).
+    echo systemd-sysusers
     # Return 0 to include the dependent module(s) in the initramfs.
     return 0
 
