diff -Naur dracut-036/modules.d/05busybox/module-setup.sh dracut-036.tpg/modules.d/05busybox/module-setup.sh
--- dracut-036/modules.d/05busybox/module-setup.sh	2014-01-29 07:35:47.000000000 +0000
+++ dracut-036.tpg/modules.d/05busybox/module-setup.sh	2014-02-12 19:32:34.762868143 +0000
@@ -4,7 +4,7 @@
 
 # called by dracut
 check() {
-    type -P busybox >/dev/null || return 1
+    type -P busybox >/dev/null || type -P busybox.static >/dev/null || return 1
 
     return 255
 }
@@ -17,7 +17,7 @@
 # called by dracut
 install() {
     local _i _progs _path _busybox
-    _busybox=$(type -P busybox)
+    _busybox=$(type -P busybox || type -P busybox.static)
     inst $_busybox /usr/bin/busybox
     for _i in $($_busybox | sed -ne '1,/Currently/!{s/,//g; s/busybox//g; p}')
     do
