diff -up dracut-051/modules.d/05busybox/module-setup.sh.6~ dracut-051/modules.d/05busybox/module-setup.sh
--- dracut-051/modules.d/05busybox/module-setup.sh.6~	2020-12-14 15:34:35.000000000 +0100
+++ dracut-051/modules.d/05busybox/module-setup.sh	2020-12-14 20:36:49.077959481 +0100
@@ -2,7 +2,7 @@
 
 # called by dracut
 check() {
-    require_binaries busybox || return 1
+    require_binaries busybox || require_binaries busybox.static || return 1
 
     return 255
 }
@@ -16,7 +16,7 @@ depends() {
 install() {
     local _i _path _busybox
     local _progs=()
-    _busybox=$(type -P busybox)
+    _busybox=$(type -P busybox || type -P busybox.static)
     inst $_busybox /usr/bin/busybox
     for _i in $($_busybox --list); do
         [[ ${_i} == busybox ]] && continue
