diff -Naur dracut-040/dracut-catimages.sh dracut-040.tpg/dracut-catimages.sh
--- dracut-040/dracut-catimages.sh	2014-10-24 11:19:58.000000000 +0000
+++ dracut-040.tpg/dracut-catimages.sh	2014-11-02 12:09:47.000000000 +0000
@@ -104,7 +104,7 @@
     ofile="$imagedir/90-overlay.img"
     dinfo "Creating image $ofile from directory $overlay"
     type pigz &>/dev/null && gzip=pigz || gzip=gzip
-    ( cd "$overlay"; find . |cpio --quiet -H newc -o |$gzip -9 > "$ofile"; )
+    ( cd "$overlay"; find . |cpio -o --quiet -H newc |$gzip -9 > "$ofile"; )
 fi
 
 if [[ ! $no_imagedir ]]; then
diff -Naur dracut-040/dracut.sh dracut-040.tpg/dracut.sh
--- dracut-040/dracut.sh	2014-11-02 11:52:38.000000000 +0000
+++ dracut-040.tpg/dracut.sh	2014-11-02 12:11:45.000000000 +0000
@@ -1594,14 +1594,14 @@
     (
         cd "$early_cpio_dir/d"
         find . -print0 | sort -z \
-            | cpio ${CPIO_REPRODUCIBLE:+--reproducible} --null $cpio_owner_root -H newc -o --quiet > $outfile
+            | cpio -o ${CPIO_REPRODUCIBLE:+--reproducible} --null $cpio_owner_root -H newc --quiet > $outfile
     )
 fi
 
 if ! (
         umask 077; cd "$initdir"
         find . -print0 | sort -z \
-            | cpio ${CPIO_REPRODUCIBLE:+--reproducible} --null $cpio_owner_root -H newc -o --quiet \
+            | cpio -o ${CPIO_REPRODUCIBLE:+--reproducible} --null $cpio_owner_root -H newc --quiet \
             | $compress >> "$outfile"
     ); then
     dfatal "dracut: creation of $outfile failed"
