diff -Naur dracut-053/modules.d/05busybox/module-setup.sh dracut-053.tpg/modules.d/05busybox/module-setup.sh
--- dracut-053/modules.d/05busybox/module-setup.sh	2021-02-23 15:11:01.000000000 +0000
+++ dracut-053.tpg/modules.d/05busybox/module-setup.sh	2021-02-25 09:48:10.624000000 +0000
@@ -2,7 +2,7 @@
 
 # called by dracut
 check() {
-    require_binaries busybox || return 1
+    require_binaries busybox || require_binaries busybox.static || return 1
 
     return 255
 }
@@ -16,7 +16,7 @@
 install() {
     local _i _path _busybox
     local _progs=()
-    _busybox=$(find_binary busybox)
+    _busybox=$(type -P busybox || type -P busybox.static)
     inst $_busybox /usr/bin/busybox
     for _i in $($_busybox --list); do
         [[ ${_i} == busybox ]] && continue
