diff -Naur dracut-007.orig//modules.d/99aufs-mount/aufs-mount.sh dracut-007/modules.d/99aufs-mount/aufs-mount.sh
--- dracut-007.orig//modules.d/99aufs-mount/aufs-mount.sh	1969-12-31 19:00:00.000000000 -0500
+++ dracut-007/modules.d/99aufs-mount/aufs-mount.sh	2011-02-09 08:30:49.000000000 -0500
@@ -0,0 +1,69 @@
+#!/bin/sh
+
+. /lib/dracut-lib.sh
+
+data=$(getarg data)
+
+if [ ! -z "$data" ] ; then
+
+UUID=${data#UUID=}
+
+mount -t tmpfs tmpfs /sysroot/tmp 
+mkdir /sysroot/tmp/xinos
+mkdir /sysroot/tmp/sysroot-rw
+mkdir /sysroot/tmp/sysroot-ro
+mount -o bind /sysroot /sysroot/tmp/sysroot-ro
+
+
+RES=0
+
+if [ ! -z "$UUID" ] ; then
+    /sbin/fsck.ext2 -y /dev/disk/by-uuid/$UUID  
+    mount /dev/disk/by-uuid/$UUID /sysroot/tmp/sysroot-rw && RES=1
+fi
+
+if [ "$RES" = 0 ] ; then 
+    mount -n -t tmpfs tmpfs /sysroot/tmp/sysroot-rw
+fi
+
+
+SKIP_DIRS="dev lost+found proc sys tmp"
+# SKIP_DIRS="bin"
+for d in `ls /sysroot `
+do
+    skip_dir=0
+    if [ ! -d "/sysroot/$d" ] ; then
+	continue
+    fi
+    for sd in $SKIP_DIRS
+    do
+        if [ $d = $sd ] ; then
+            skip_dir=1
+        fi
+    done
+    if [ $skip_dir = 1 ] ; then
+        continue
+    fi
+
+    if [ ! -e "/sysroot/tmp/sysroot-rw/$d" ] ; then
+	if [ $d = "root" ] ; then
+	    mkdir -m 750 "/sysroot/tmp/sysroot-rw/$d"
+	else
+	    mkdir -m 755 "/sysroot/tmp/sysroot-rw/$d"
+	fi
+    fi
+    if [ ! -d "/sysroot/tmp/sysroot-rw/$d" ] ; then 
+	continue
+    fi
+
+    mount -n -t aufs -o nowarn_perm,noatime,xino=/sysroot/tmp/xinos/$d.xino.aufs,dirs=/sysroot/tmp/sysroot-rw/$d=rw:/sysroot/$d=rr none /sysroot/$d
+
+done
+
+# cp -f /etc/halt /sysroot/etc/init.d/halt
+# cp -f /etc/functions /sysroot/etc/init.d/functions
+
+# mount -n -t aufs -o noatime,xino=/xinos/sysroot.xino.aufs,dirs=/sysroot-rw=rw:/sysroot=rr none /sysroot
+
+fi # if [ ! -z "$data" ]
+
diff -Naur dracut-007.orig//modules.d/99aufs-mount/install dracut-007/modules.d/99aufs-mount/install
--- dracut-007.orig//modules.d/99aufs-mount/install	1969-12-31 19:00:00.000000000 -0500
+++ dracut-007/modules.d/99aufs-mount/install	2011-02-09 08:30:08.000000000 -0500
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+dracut_install fsck.ext2
+
+inst_hook pre-pivot 10 "$moddir/aufs-mount.sh"
