--- a/modules.d/00systemd/module-setup.sh.omv~	2017-05-15 23:55:18.602776214 +0200
+++ b/modules.d/00systemd/module-setup.sh	2017-05-16 00:01:29.432360533 +0200
@@ -200,6 +200,7 @@ install() {
     inst_binary true
     ln_r $(type -P true) "/usr/bin/loginctl"
     ln_r $(type -P true) "/bin/loginctl"
+    [ "`uname -m`" = "aarch64" ] && ln_r /lib64/ld-linux-aarch64.so.1 "/lib/ld-linux-aarch64.so.1"
     inst_rules \
         70-uaccess.rules \
         71-seat.rules \
