diff -Nur dracut-024.old/modules.d/50plymouth/plymouth-populate-initrd.sh dracut-024/modules.d/50plymouth/plymouth-populate-initrd.sh
--- dracut-024.old/modules.d/50plymouth/plymouth-populate-initrd.sh	2012-10-16 18:31:06.000000000 +0400
+++ dracut-024/modules.d/50plymouth/plymouth-populate-initrd.sh	2012-11-27 16:34:07.684822928 +0400
@@ -1,13 +1,13 @@
 #!/bin/bash
 # -*- mode: shell-script; indent-tabs-mode: nil; sh-basic-offset: 4; -*-
 # ex: ts=8 sw=4 sts=4 et filetype=sh
-PLYMOUTH_LOGO_FILE="/usr/share/pixmaps/system-logo-white.png"
+PLYMOUTH_LOGO_FILE="/usr/share/plymouth/themes/Mandriva-*/welcome.png"
 PLYMOUTH_THEME=$(plymouth-set-default-theme)
 
 inst /sbin/plymouthd /bin/plymouthd
 dracut_install /bin/plymouth \
     "${PLYMOUTH_LOGO_FILE}" \
-    /etc/system-release
+    /etc/rosa-release
 
 mkdir -m 0755 -p "${initdir}/usr/share/plymouth"
 
diff -Nur dracut-024.old/modules.d/90dm/dm-pre-udev.sh dracut-024/modules.d/90dm/dm-pre-udev.sh
--- dracut-024.old/modules.d/90dm/dm-pre-udev.sh	2012-10-16 18:31:06.000000000 +0400
+++ dracut-024/modules.d/90dm/dm-pre-udev.sh	2012-11-27 16:34:53.817826138 +0400
@@ -1,5 +1,5 @@
 #!/bin/sh
 # -*- mode: shell-script; indent-tabs-mode: nil; sh-basic-offset: 4; -*-
 # ex: ts=8 sw=4 sts=4 et filetype=sh
-strstr "$(cat /proc/misc)" device-mapper || modprobe dm_mod
+strstr "$(cat /proc/misc)" device-mapper || modprobe dm-mod
 modprobe dm_mirror 2>/dev/null
diff -Nur dracut-024.old/modules.d/90dmsquash-live/dmsquash-live-genrules.sh dracut-024/modules.d/90dmsquash-live/dmsquash-live-genrules.sh
--- dracut-024.old/modules.d/90dmsquash-live/dmsquash-live-genrules.sh	2012-10-16 18:31:06.000000000 +0400
+++ dracut-024/modules.d/90dmsquash-live/dmsquash-live-genrules.sh	2012-11-27 16:35:21.954827788 +0400
@@ -12,6 +12,7 @@
     wait_for_dev "${root#live:}"
   ;;
   live:*)
+  modprobe loop
     if [ -f "${root#live:}" ]; then
         /sbin/initqueue --settled --onetime --unique /sbin/dmsquash-live-root "${root#live:}"
     fi
diff -Nur dracut-024.old/modules.d/90dmsquash-live/dmsquash-liveiso-genrules.sh dracut-024/modules.d/90dmsquash-live/dmsquash-liveiso-genrules.sh
--- dracut-024.old/modules.d/90dmsquash-live/dmsquash-liveiso-genrules.sh	2012-10-16 18:31:06.000000000 +0400
+++ dracut-024/modules.d/90dmsquash-live/dmsquash-liveiso-genrules.sh	2012-11-27 16:35:50.239828815 +0400
@@ -1,6 +1,7 @@
 #!/bin/sh
 # -*- mode: shell-script; indent-tabs-mode: nil; sh-basic-offset: 4; -*-
 # ex: ts=8 sw=4 sts=4 et filetype=sh
+modprobe loop
 if [ "${root%%:*}" = "liveiso" ]; then
     {
         printf 'KERNEL=="loop0", RUN+="/sbin/initqueue --settled --onetime --unique /sbin/dmsquash-live-root `/sbin/losetup -f --show %s`"\n' \
