diff -up dracut-ng-108/modules.d/70lvm/module-setup.sh.2~ dracut-ng-108/modules.d/70lvm/module-setup.sh
--- dracut-ng-108/modules.d/70lvm/module-setup.sh.2~	2025-08-04 12:40:12.000000000 +0200
+++ dracut-ng-108/modules.d/70lvm/module-setup.sh	2025-09-05 23:42:07.479039541 +0200
@@ -89,6 +89,8 @@ install() {
     inst_script "$moddir/lvm_scan.sh" /sbin/lvm_scan
     inst_hook cmdline 30 "$moddir/parse-lvm.sh"
 
+    inst_libdir_file "libdevmapper-event*.so*"
+
     if [[ ${hostonly-} ]] && find_binary lvs &> /dev/null; then
         for dev in "${!host_fs_types[@]}"; do
             [[ -e /sys/block/${dev#/dev/}/dm/name ]] || continue
