diff -up dracut-059/modules.d/90lvm/module-setup.sh.omv~ dracut-059/modules.d/90lvm/module-setup.sh
--- dracut-059/modules.d/90lvm/module-setup.sh.omv~	2023-05-12 23:23:00.026116424 +0200
+++ dracut-059/modules.d/90lvm/module-setup.sh	2023-05-12 23:23:15.613263457 +0200
@@ -89,6 +89,8 @@ install() {
     inst_script "$moddir/lvm_scan.sh" /sbin/lvm_scan
     inst_hook cmdline 30 "$moddir/parse-lvm.sh"
 
+    inst_libdir_file "libdevmapper-event*.so*"
+
     if [[ $hostonly ]] && find_binary lvs &> /dev/null; then
         for dev in "${!host_fs_types[@]}"; do
             [[ -e /sys/block/${dev#/dev/}/dm/name ]] || continue
