diff -Naur dracut-049/modules.d/99fs-lib/module-setup.sh dracut-049.tpg/modules.d/99fs-lib/module-setup.sh
--- dracut-049/modules.d/99fs-lib/module-setup.sh	2018-10-08 13:38:33.000000000 +0000
+++ dracut-049.tpg/modules.d/99fs-lib/module-setup.sh	2019-01-08 11:15:45.492328780 +0000
@@ -69,7 +69,7 @@
 
     if [[ "$fscks" = "${fscks#*[^ ]*}" ]]; then
         _helpers="\
-            umount mount /sbin/fsck*
+            umount mount /sbin/fsck* /usr/sbin/fsck*
             xfs_db xfs_check xfs_repair xfs_metadump
             e2fsck jfs_fsck reiserfsck btrfsck
         "
