--- dracut-027/dracut-functions.sh	2013-03-26 15:55:03.000000000 +0000
+++ dracut-027/dracut-functions.sh.tpg	2013-03-30 18:37:41.378207799 +0000
@@ -1461,6 +1461,7 @@
 
 dracut_kernel_post() {
     local _moddirname=${srcmods%%/lib/modules/*}
+    local _compress=$1
 
     if [[ $DRACUT_KERNEL_LAZY_HASHDIR ]] && [[ -f "$DRACUT_KERNEL_LAZY_HASHDIR/lazylist" ]]; then
         xargs -r modprobe -a ${_moddirname+-d ${_moddirname}/} \
@@ -1517,6 +1518,15 @@
         [[ $srcmods/$_f ]] && inst_simple "$srcmods/$_f" "/lib/modules/$kernel/$_f"
     done
 
+    # we get better compression ratio if we don't compress files twice, while also
+    # saving ourself from the overhead of decompressing twice as well..
+    for mod in $(find ${initdir}${srcmods} -type f); do
+        case ${mod} in
+            *.ko.gz) gunzip ${mod} ;;
+            *.ko.xz) unxz ${mod} ;;
+        esac
+    done
+
     # generate module dependencies for the initrd
     if [[ -d $initdir/lib/modules/$kernel ]] && \
         ! depmod -a -b "$initdir" $kernel; then
--- dracut-027/dracut.sh	2013-03-26 15:55:03.000000000 +0000
+++ dracut-027/dracut.sh.tpg	2013-03-30 18:38:44.334203255 +0000
@@ -1040,7 +1040,7 @@
     fi
 
     dinfo "*** Installing kernel module dependencies and firmware ***"
-    dracut_kernel_post
+    dracut_kernel_post $compress
     dinfo "*** Installing kernel module dependencies and firmware done ***"
 
     if [[ $noimageifnotneeded == yes ]] && [[ $hostonly ]]; then
