diff -Naur dracut-037/modules.d/98systemd/module-setup.sh dracut-037.tpg/modules.d/98systemd/module-setup.sh
--- dracut-037/modules.d/98systemd/module-setup.sh	2014-03-19 16:16:08.000000000 +0000
+++ dracut-037.tpg/modules.d/98systemd/module-setup.sh	2014-04-18 15:42:24.141190084 +0000
@@ -162,6 +162,11 @@
 
         _mods=$(modules_load_get /etc/modules-load.d)
         [[ $_mods ]] && instmods $_mods
+# (tpg) this is due bug https://issues.openmandriva.org/show_bug.cgi?id=669
+    else
+	inst_multiple -o \
+	/etc/vconsole.conf \
+	/etc/locale.conf
     fi
 
     if ! [[ -e "$initdir/etc/machine-id" ]]; then
