diff -Naur dracut-044/dracut.sh dracut-044.tpg/dracut.sh
--- dracut-044/dracut.sh	2016-03-01 20:08:06.000000000 +0000
+++ dracut-044.tpg/dracut.sh	2016-03-01 20:09:12.019255706 +0000
@@ -1591,7 +1591,7 @@
     # strip kernel modules, but do not touch signed modules
     find "$initdir" -type f -path '*/lib/modules/*.ko' -print0 \
         | while read -r -d $'\0' f || [ -n "$f" ]; do
-        SIG=$(tail -c 28 "$f")
+        SIG=$(tail -c 28 "$f" | tr -d '\000')
         [[ $SIG == '~Module signature appended~' ]] || { printf "%s\000" "$f"; }
     done | xargs -r -0 strip -g
 
diff -Naur dracut-044/modules.d/50drm/module-setup.sh dracut-044.tpg/modules.d/50drm/module-setup.sh
--- dracut-044/modules.d/50drm/module-setup.sh	2015-11-25 13:22:28.000000000 +0000
+++ dracut-044.tpg/modules.d/50drm/module-setup.sh	2016-03-01 20:09:43.319105594 +0000
@@ -24,9 +24,9 @@
             local _fname _fcont
             while read _fname || [ -n "$_fname" ]; do
                 case "$_fname" in
-                    *.ko)    _fcont="$(<        $_fname)" ;;
-                    *.ko.gz) _fcont="$(gzip -dc $_fname)" ;;
-                    *.ko.xz) _fcont="$(xz -dc   $_fname)" ;;
+                    *.ko)    _fcont="$(<        "$_fname" | tr -d '\000')" ;;
+                    *.ko.gz) _fcont="$(gzip -dc "$_fname" | tr -d '\000')" ;;
+                    *.ko.xz) _fcont="$(xz -dc   "$_fname" | tr -d '\000')" ;;
                 esac
                 [[   $_fcont =~ $_drm_drivers
                 && ! $_fcont =~ iw_handler_get_spy ]] \
diff -Naur dracut-044/modules.d/90kernel-modules/module-setup.sh dracut-044.tpg/modules.d/90kernel-modules/module-setup.sh
--- dracut-044/modules.d/90kernel-modules/module-setup.sh	2015-11-25 13:22:28.000000000 +0000
+++ dracut-044.tpg/modules.d/90kernel-modules/module-setup.sh	2016-03-01 20:10:12.798906099 +0000
@@ -10,9 +10,9 @@
             function bmf1() {
                 local _f
                 while read _f || [ -n "$_f" ]; do case "$_f" in
-                    *.ko)    [[ $(<         $_f) =~ $_blockfuncs ]] && echo "$_f" ;;
-                    *.ko.gz) [[ $(gzip -dc <$_f) =~ $_blockfuncs ]] && echo "$_f" ;;
-                    *.ko.xz) [[ $(xz -dc   <$_f) =~ $_blockfuncs ]] && echo "$_f" ;;
+                    *.ko)    [[ $(<         "$_f" | tr -d '\000') =~ $_blockfuncs ]] && echo "$_f" ;;
+                    *.ko.gz) [[ $(gzip -dc <"$_f" | tr -d '\000') =~ $_blockfuncs ]] && echo "$_f" ;;
+                    *.ko.xz) [[ $(xz -dc   <"$_f" | tr -d '\000') =~ $_blockfuncs ]] && echo "$_f" ;;
                     esac
                 done
                 return 0
