diff -Naur dracut-036/modules.d/05busybox/module-setup.sh dracut-036.tpg/modules.d/05busybox/module-setup.sh
--- dracut-036/modules.d/05busybox/module-setup.sh	2014-02-12 19:40:01.914830798 +0000
+++ dracut-036.tpg/modules.d/05busybox/module-setup.sh	2014-02-12 19:40:45.037992632 +0000
@@ -19,7 +19,7 @@
     local _i _progs _path _busybox
     _busybox=$(type -P busybox || type -P busybox.static)
     inst $_busybox /usr/bin/busybox
-    for _i in $($_busybox | sed -ne '1,/Currently/!{s/,//g; s/busybox//g; p}')
+    for _i in $($_busybox --list)
     do
         _progs="$_progs $_i"
     done
