diff -up dracut-ng-109/modules.d/70lvm/module-setup.sh.2~ dracut-ng-109/modules.d/70lvm/module-setup.sh
--- dracut-ng-109/modules.d/70lvm/module-setup.sh.2~	2025-11-03 23:35:34.000000000 +0100
+++ dracut-ng-109/modules.d/70lvm/module-setup.sh	2025-11-04 17:21:52.217548263 +0100
@@ -89,6 +89,8 @@ install() {
     inst_script "$moddir/lvm_scan.sh" /sbin/lvm_scan
     inst_hook cmdline 30 "$moddir/parse-lvm.sh"
 
+    inst_libdir_file "libdevmapper-event*.so*"
+
     if [[ $hostonly ]] && find_binary lvs &> /dev/null; then
         for dev in "${!host_fs_types[@]}"; do
             [[ -e /sys/block/${dev#/dev/}/dm/name ]] || continue
