diff -Naur dracut-053/dracut-init.sh dracut-053.tpg/dracut-init.sh
--- dracut-053/dracut-init.sh	2021-02-23 15:11:01.000000000 +0000
+++ dracut-053.tpg/dracut-init.sh	2021-02-25 09:50:58.758000000 +0000
@@ -986,6 +986,16 @@
         [[ -e $srcmods/$_f ]] && inst_simple "$srcmods/$_f" "/lib/modules/$kernel/$_f"
     done
 
+    # we get better compression ratio if we don't compress files twice, while also
+    # saving ourself from the overhead of decompressing twice as well..
+    for mod in $(find ${initdir}${srcmods} -type f); do
+        case ${mod} in
+            *.ko.gz) gunzip ${mod} ;;
+            *.ko.xz) unxz -T0 ${mod} ;;
+            *.ko.zst) unzstd ${mod} -q --rm -f ;;
+        esac
+    done
+
     # generate module dependencies for the initrd
     if [[ -d $initdir/lib/modules/$kernel ]] && \
         ! depmod -a -b "$initdir" "$kernel"; then
