diff -Naur dracut-040/modules.d/90dmsquash-live/dmsquash-live-genrules.sh dracut-040.tpg/modules.d/90dmsquash-live/dmsquash-live-genrules.sh
--- dracut-040/modules.d/90dmsquash-live/dmsquash-live-genrules.sh	2014-10-24 11:19:58.000000000 +0000
+++ dracut-040.tpg/modules.d/90dmsquash-live/dmsquash-live-genrules.sh	2014-11-02 11:57:41.269586867 +0000
@@ -11,6 +11,7 @@
     wait_for_dev -n "${root#live:}"
   ;;
   live:*)
+  modprobe loop
     if [ -f "${root#live:}" ]; then
         /sbin/initqueue --settled --onetime --unique /sbin/dmsquash-live-root "${root#live:}"
     fi
diff -Naur dracut-040/modules.d/90dmsquash-live/dmsquash-liveiso-genrules.sh dracut-040.tpg/modules.d/90dmsquash-live/dmsquash-liveiso-genrules.sh
--- dracut-040/modules.d/90dmsquash-live/dmsquash-liveiso-genrules.sh	2014-10-24 11:19:58.000000000 +0000
+++ dracut-040.tpg/modules.d/90dmsquash-live/dmsquash-liveiso-genrules.sh	2014-11-02 11:58:44.478329585 +0000
@@ -1,5 +1,6 @@
 #!/bin/sh
 
+modprobe loop
 if [ "${root%%:*}" = "liveiso" ]; then
     {
         printf 'KERNEL=="loop-control", RUN+="/sbin/initqueue --settled --onetime --unique /sbin/dmsquash-live-root `/sbin/losetup -f --show %s`"\n' \
