diff -Naur dracut-045/dracut-init.sh dracut-045.tpg/dracut-init.sh
--- dracut-045/dracut-init.sh	2017-04-03 08:26:25.000000000 +0000
+++ dracut-045.tpg/dracut-init.sh	2017-04-15 11:04:04.007344797 +0000
@@ -936,12 +936,22 @@
 
 dracut_kernel_post() {
     local _moddirname=${srcmods%%/lib/modules/*}
+    local _compress=$1
     local _pid
 
     for _f in modules.builtin.bin modules.builtin modules.order; do
         [[ $srcmods/$_f ]] && inst_simple "$srcmods/$_f" "/lib/modules/$kernel/$_f"
     done
 
+    # we get better compression ratio if we don't compress files twice, while also
+    # saving ourself from the overhead of decompressing twice as well..
+    for mod in $(find ${initdir}${srcmods} -type f); do
+        case ${mod} in
+            *.ko.gz) gunzip ${mod} ;;
+            *.ko.xz) unxz -T0 ${mod} ;;
+        esac
+    done
+
     # generate module dependencies for the initrd
     if [[ -d $initdir/lib/modules/$kernel ]] && \
         ! depmod -a -b "$initdir" $kernel; then
