diff -Naur dracut-022/mkinitrd-dracut.sh dracut-022.tpg/mkinitrd-dracut.sh
--- dracut-022/mkinitrd-dracut.sh	2012-07-20 11:11:21.000000000 +0000
+++ dracut-022.tpg/mkinitrd-dracut.sh	2012-07-22 15:55:14.000000000 +0000
@@ -8,10 +8,18 @@
 
     $cmd "usage: ${0##*/} [--version] [--help] [-v] [-f] [--preload <module>]"
     $cmd "       [--image-version] [--with=<module>]"
+    $cmd "       [--force-scsi-probe | --omit-scsi-modules]"
+    $cmd "       [--omit-ide-modules]"
+    $cmd "       [--force-raid-probe | --omit-raid-modules]"
+    $cmd "       [--force-lvm-probe | --omit-lvm-modules]"
+    $cmd "       [--builtin=<module>] [--omit-dmraid] [--net-dev=<interface>]"
+    $cmd "       [--fstab=<fstab>] [--nocompress]"
+    $cmd "       [--dsdt[=<dsdt.aml>]] [--bootchart]]"
     $cmd "       [--nocompress]"
     $cmd "       <initrd-image> <kernel-version>"
     $cmd ""
     $cmd "       (ex: ${0##*/} /boot/initramfs-$kver.img $kver)"
+    $cmd "       (ex: ${0##*/} /boot/initrd-$(uname -r).img $kver)"
 
     [[ $1 = '-n' ]] && exit 0
     exit 1
