diff -Naur dracut-047/dracut.sh dracut-047.tpg/dracut.sh
--- dracut-047/dracut.sh	2018-02-23 20:18:55.445924000 +0000
+++ dracut-047.tpg/dracut.sh	2018-02-23 20:23:48.914318775 +0000
@@ -1761,7 +1761,7 @@
     if ! (
             umask 077; cd "$early_cpio_dir/d"
             find . -print0 | sort -z \
-                | cpio ${CPIO_REPRODUCIBLE:+--reproducible} --null $cpio_owner_root -H newc -o --quiet > "${DRACUT_TMPDIR}/initramfs.img"
+                | cpio -o ${CPIO_REPRODUCIBLE:+--reproducible} --null $cpio_owner_root -H newc --quiet > "${DRACUT_TMPDIR}/initramfs.img"
         ); then
         dfatal "dracut: creation of $outfile failed"
         exit 1
@@ -1771,7 +1771,7 @@
 if ! (
         umask 077; cd "$initdir"
         find . -print0 | sort -z \
-            | cpio ${CPIO_REPRODUCIBLE:+--reproducible} --null $cpio_owner_root -H newc -o --quiet \
+            | cpio -o ${CPIO_REPRODUCIBLE:+--reproducible} --null $cpio_owner_root -H newc --quiet \
             | $compress >> "${DRACUT_TMPDIR}/initramfs.img"
     ); then
     dfatal "dracut: creation of $outfile failed"
