diff -Naur dracut-046/dracut-init.sh dracut-046.tpg/dracut-init.sh
--- dracut-046/dracut-init.sh	2017-08-11 11:44:06.000000000 +0000
+++ dracut-046.tpg/dracut-init.sh	2017-09-04 21:57:05.201409486 +0000
@@ -939,6 +939,15 @@
         [[ $srcmods/$_f ]] && inst_simple "$srcmods/$_f" "/lib/modules/$kernel/$_f"
     done
 
+    # we get better compression ratio if we don't compress files twice, while also
+    # saving ourself from the overhead of decompressing twice as well..
+    for mod in $(find ${initdir}${srcmods} -type f); do
+        case ${mod} in
+            *.ko.gz) gunzip ${mod} ;;
+            *.ko.xz) unxz -T0 ${mod} ;;
+        esac
+    done
+
     # generate module dependencies for the initrd
     if [[ -d $initdir/lib/modules/$kernel ]] && \
         ! depmod -a -b "$initdir" $kernel; then
