diff -Naur dracut-036/dracut.sh dracut-036.tpg/dracut.sh
--- dracut-036/dracut.sh	2014-01-29 07:35:47.000000000 +0000
+++ dracut-036.tpg/dracut.sh	2014-04-08 22:09:23.318100032 +0000
@@ -603,7 +603,7 @@
     if [[ $MACHINE_ID ]] && ( [[ -d /boot/${MACHINE_ID} ]] || [[ -L /boot/${MACHINE_ID} ]] ); then
         outfile="/boot/${MACHINE_ID}/$kernel/initrd"
     else
-        outfile="/boot/initramfs-$kernel.img"
+        outfile="/boot/initrd-$kernel.img"
     fi
 fi
 
