diff -Naur dracut-037/modules.d/03modsign/module-setup.sh dracut-037.tpg/modules.d/03modsign/module-setup.sh
--- dracut-037/modules.d/03modsign/module-setup.sh	2014-03-19 16:16:08.000000000 +0000
+++ dracut-037.tpg/modules.d/03modsign/module-setup.sh	2014-04-15 21:51:55.636711149 +0000
@@ -29,7 +29,7 @@
 # called by dracut
 install() {
     inst_dir /lib/modules/keys
-    inst_binary /usr/bin/keyctl
+    inst_binary /bin/keyctl
 
     inst_hook pre-trigger 01 "$moddir/load-modsign-keys.sh"
 
