diff -Naur dracut-048/dracut-init.sh dracut-048.tpg/dracut-init.sh
--- dracut-048/dracut-init.sh	2018-07-06 08:37:51.000000000 +0000
+++ dracut-048.tpg/dracut-init.sh	2018-07-12 13:14:11.908773703 +0000
@@ -957,6 +957,16 @@
         [[ -e $srcmods/$_f ]] && inst_simple "$srcmods/$_f" "/lib/modules/$kernel/$_f"
     done
 
+    # we get better compression ratio if we don't compress files twice, while also
+    # saving ourself from the overhead of decompressing twice as well..
+    for mod in $(find ${initdir}${srcmods} -type f); do
+        case ${mod} in
+            *.ko.gz) gunzip ${mod} ;;
+            *.ko.xz) unxz -T0 ${mod} ;;
+            *.ko.zst) unzstd ${mod} -q --rm -f ;;
+        esac
+    done
+
     # generate module dependencies for the initrd
     if [[ -d $initdir/lib/modules/$kernel ]] && \
         ! depmod -a -b "$initdir" $kernel; then
