diff -Naur dracut-031/dracut-functions.sh dracut-031.tpg/dracut-functions.sh
--- dracut-031/dracut-functions.sh	2013-07-31 12:24:25.000000000 +0000
+++ dracut-031.tpg/dracut-functions.sh	2013-08-01 08:01:50.751177240 +0000
@@ -1468,6 +1468,7 @@
 
 dracut_kernel_post() {
     local _moddirname=${srcmods%%/lib/modules/*}
+    local _compress=$1
     local _pid
 
     if [[ $DRACUT_KERNEL_LAZY_HASHDIR ]] && [[ -f "$DRACUT_KERNEL_LAZY_HASHDIR/lazylist" ]]; then
@@ -1527,6 +1528,15 @@
         [[ $srcmods/$_f ]] && inst_simple "$srcmods/$_f" "/lib/modules/$kernel/$_f"
     done
 
+    # we get better compression ratio if we don't compress files twice, while also
+    # saving ourself from the overhead of decompressing twice as well..
+    for mod in $(find ${initdir}${srcmods} -type f); do
+        case ${mod} in
+            *.ko.gz) gunzip ${mod} ;;
+            *.ko.xz) unxz ${mod} ;;
+        esac
+    done
+
     # generate module dependencies for the initrd
     if [[ -d $initdir/lib/modules/$kernel ]] && \
         ! depmod -a -b "$initdir" $kernel; then
diff -Naur dracut-031/dracut.sh dracut-031.tpg/dracut.sh
--- dracut-031/dracut.sh	2013-07-31 12:24:25.000000000 +0000
+++ dracut-031.tpg/dracut.sh	2013-08-01 08:00:55.754634687 +0000
@@ -1089,7 +1089,7 @@
     fi
 
     dinfo "*** Installing kernel module dependencies and firmware ***"
-    dracut_kernel_post
+    dracut_kernel_post $compress
     dinfo "*** Installing kernel module dependencies and firmware done ***"
 
     if [[ $noimageifnotneeded == yes ]] && [[ $hostonly ]]; then
