--- dracut-045/modules.d/90dmsquash-live/dmsquash-live-root.sh	2017-04-03 09:26:25.000000000 +0100
+++ dracut-045/modules.d/90dmsquash-live/dmsquash-live-root.sh~	2017-09-02 14:00:34.158699893 +0100
@@ -166,9 +166,8 @@
         elif [ -n "$devspec" -a -n "$pathspec" ]; then
             [ -z "$m" ] &&
                 m='  Unable to find a persistent overlay; using a temporary one.'
-            m=($'\n' "$m" $'\n'
-               '     All root filesystem changes will be lost on shutdown.'
-               $'\n' '        Press any key to continue')
+		m=$(printf '%s\n' "$m" "     All root filesystem changes will be lost on shutdown." \
+             "        Press any key to continue"
             echo -e "\n\n\n${m[*]}\n\n\n" > /dev/kmsg
             if [ -n "$DRACUT_SYSTEMD" ]; then
                 if plymouth --ping ; then
